src/js/TuningSliders.js

```diff
@@ -414,61 +449,97 @@ TuningSliders.calculateNewPids = function() {
-    $('.pid_filter select[name="gyroLowpass2Type').val(FC.FILTER_CONFIG.gyro_lowpass2_type);
-    $('output[name="sliderGyroFilterMultiplier-number"]').val(this.sliderGyroFilterMultiplier);
+    this.updateLowpassValues();
 };
 
-TuningSliders.calculateNewDTermFilters = function() {
+TuningSliders.calculateLegacyDTermFilters = function() {
     // calculate, set and display new values in forms based on slider position
     FC.FILTER_CONFIG.dterm_lowpass_dyn_min_hz = Math.floor(this.FILTER_DEFAULT.dterm_lowpass_dyn_min_hz * this.sliderDTermFilterMultiplier);
     FC.FILTER_CONFIG.dterm_lowpass_dyn_max_hz = Math.floor(this.FILTER_DEFAULT.dterm_lowpass_dyn_max_hz * this.sliderDTermFilterMultiplier);
     FC.FILTER_CONFIG.dterm_lowpass2_hz = Math.floor(this.FILTER_DEFAULT.dterm_lowpass2_hz * this.sliderDTermFilterMultiplier);
     FC.FILTER_CONFIG.dterm_lowpass_type = this.FILTER_DEFAULT.dterm_lowpass_type;
     FC.FILTER_CONFIG.dterm_lowpass2_type = this.FILTER_DEFAULT.dterm_lowpass2_type;
 
+    this.updateLowpassValues();
+};
+
+TuningSliders.calculateNewGyroFilters = function() {
+    // this is the main calculation for Gyro Filter slider, inputs are in form of slider position values
+    // values get set both into forms and their respective variables
+    if (semver.lte(FC.CONFIG.apiVersion, API_VERSION_1_43)) {
+        this.calculateLegacyGyroFilters();
+
+    }
+
+    if (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_44)) {
+        FC.TUNING_SLIDERS.slider_gyro_filter_multiplier = this.sliderGyroFilterMultiplier * 100;
+
+        Promise.resolve(true)
+        .then(() => { return MSP.promise(MSPCodes.MSP_SET_TUNING_SLIDERS, mspHelper.crunch(MSPCodes.MSP_SET_TUNING_SLIDERS)); })
+        .then(() => { return MSP.send_message(MSPCodes.MSP_FILTER_CONFIG); });
+
+        const delayInMilliseconds = 250; //0.25 seconds
+        setTimeout(function() {
+            TuningSliders.updateLowpassValues();
+        }, delayInMilliseconds);
+    }
+};
+
+TuningSliders.calculateNewDTermFilters = function() {
+    // this is the main calculation for Gyro Filter slider, inputs are in form of slider position values
+    // values get set both into forms and their respective variables
+    if (semver.lte(FC.CONFIG.apiVersion, API_VERSION_1_43)) {
+        this.calculateLegacyDTermFilters();
+
+    }
+
+    if (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_44)) {
+        FC.TUNING_SLIDERS.slider_dterm_filter_multiplier = this.sliderDTermFilterMultiplier * 100;
+
+        Promise.resolve(true)
+        .then(() => { return MSP.promise(MSPCodes.MSP_SET_TUNING_SLIDERS, mspHelper.crunch(MSPCodes.MSP_SET_TUNING_SLIDERS)); })
+        .then(() => { return MSP.send_message(MSPCodes.MSP_FILTER_CONFIG); });
+
+        const delayInMilliseconds = 250; //0.25 seconds
+        setTimeout(function() {
+            TuningSliders.updateLowpassValues();
+        }, delayInMilliseconds);
```

Comment:
Same here, `Promise.resolve` is unnecessary, and would be nice to check if we can avoid `setTimeout`

Reply:
@chmelevskij , unfortunately without the Timeout function, I'm getting this behavior.  It takes somewhere between 1ms to 2.5ms for the board to calc the new PID values and returns them to the configurator.  So we need to wait for that to be done.  I'm thinking the ".then" function is only waiting till the Configurator does the line above, which doesn't take into account the delay of the FC and USB interface.  If you know of a better way than Timeout, I'm all ears.

![2IsE8WVtmj](https://user-images.githubusercontent.com/25570978/109496407-9b8aa000-7a5e-11eb-8d6a-5f6b09b55917.gif)